my_enumerables.rb

```diff
@@ -0,0 +1,122 @@
+module Enumerable
+  def my_each
+    return enum_for(__method__) unless block_given?
+
+    arr = to_a
+    size.times { |idx| yield(arr[idx]) }
+    self
+  end
+
+  def my_each_with_index
+    return enum_for(__method__) unless block_given?
+
+    arr = to_a
+    size.times { |idx| yield(arr[idx], idx) }
+    self
+  end
+
+  def my_select
+    return enum_for(__method__) unless block_given?
+
+    result = []
+    my_each { |el| result << el if yield(el) }
+    result
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity
+  # rubocop:disable Metrics/PerceivedComplexity
+  def my_all?(pattarn_or_cls = nil)
+    my_each do |el|
+      if pattarn_or_cls
+        return false if pattarn_or_cls.is_a?(Regexp) && el !~ pattarn_or_cls
+        return false if !pattarn_or_cls.is_a?(Regexp) && pattarn_or_cls.is_a?(Object) && !el.is_a?(pattarn_or_cls)
+      elsif block_given?
+        return false unless yield(el)
+      else
+        return false unless el
+      end
+    end
+    true
+  end
+
+  def my_any?(pattarn_or_cls = nil)
+    my_each do |el|
+      if pattarn_or_cls
+        return true if pattarn_or_cls.is_a?(Regexp) && el =~ pattarn_or_cls
+        return true if !pattarn_or_cls.is_a?(Regexp) && pattarn_or_cls.is_a?(Object) && el.is_a?(pattarn_or_cls)
+      elsif block_given?
+        return true if yield(el)
+      elsif el
+        return true
+      end
+    end
+    false
+  end
```

Comment:
This method should work with Regex or a Class as a parameter.
Right now I get an error "Class or module required"
Example: `enumerable.my_any?(5)`
Example: `enumerable.my_any?('string')`